File: 1592-rearrange-spaces-between-words/main.py
class Solution:
    def reorderSpaces(self, text: str) -> str:
        
        if not text or " " not in text: 
            return text

        count_spaces = text.count(" ")
        without_spaces_list = text.split()

        if len(without_spaces_list) == 1:
            return without_spaces_list[0] + " " * count_spaces
        
        div_spaces = count_spaces // (len(without_spaces_list) - 1)
        remaining_spaces = count_spaces % (len(without_spaces_list) - 1)
        
        for i in range(len(without_spaces_list) - 1):
            without_spaces_list[i] += " " * div_spaces

        without_spaces_list[-1] += " " * remaining_spaces
    
        return "".join(without_spaces_list)

File: 1592-rearrange-spaces-between-words/main.go
import "strings"

func reorderSpaces(text string) string {
    
    if len(text) == 0 || !strings.Contains(text," ") {
        return text
    }

    countSpaces := strings.Count(text, " ")
    withoutSpacesList := strings.Fields(text)

    m := len(withoutSpacesList)

    if m  == 1 {
        return withoutSpacesList[0] + strings.Repeat(" ", countSpaces)
    }

    divSpaces := countSpaces / (m - 1)
    remainingSpaces := countSpaces % (m - 1)

    for i := range withoutSpacesList[:m-1] {
        withoutSpacesList[i] += strings.Repeat(" ", divSpaces)
    }

    withoutSpacesList[m-1] += strings.Repeat(" ", remainingSpaces)

    return strings.Join(withoutSpacesList, "")

}
